### ISSUE TITLE
WeakReferenceCleaner now removed from REFERENCE_MAP on scheduleForClean() but breaks a test

### ISSUE THREAD

**Issue by peter-lawrey** [OpenHFT/Chronicle-Bytes#68]
The following enhancement to stop a map growing large under load and causing jitter, causes a test monitoring this map to blow up.

net.openhft.chronicle.queue.impl.single.AppenderFileHandleLeakTest#tailerShouldReleaseFileHandlesAsQueueRolls

```
    public void scheduleForClean() {
        SCHEDULED_CLEAN.add(this);
        REFERENCE_MAP.remove(this); // added to stop this map blowing up.
    }
```

**Comment by RobAustin** [comment]
see 
http://teamcity.higherfrequencytrading.com:8111/viewLog.html?buildId=269176&tab=buildResultsDiv&buildTypeId=OpenHFT_ChronicleQueue4_Snapshot

**Comment by hft-team-city** [comment]
Released in Chronicle-Bytes-1.16.17 (https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-1.16.17), BOM-1.16.102 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-1.16.102)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.101](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
